geoservices/titiler_service_stack.py

```diff
@@ -0,0 +1,160 @@
+import os
+from aws_cdk import (
+    Stack,
+    BundlingOptions,
+    DockerImage,
+    Duration,
+    Fn,
+    CfnOutput,
+    aws_certificatemanager as certificatemanager,
+    aws_cloudfront as cloudfront,
+    aws_cloudfront_origins as cloudfront_origins,
+    aws_iam as iam,
+    aws_s3 as s3,
+    aws_wafv2 as waf,
+    aws_lambda,
+    aws_events,
+    aws_events_targets
+)
+
+from constructs import Construct
+
+class TitilerServiceStack(Stack):
+    def __init__(self, scope: Construct, construct_id: str, stage: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Environment Variables
+        env = {
+            "CPL_VSIL_CURL_ALLOWED_EXTENSIONS": ".tif,.TIF,.tiff",
+            "GDAL_CACHEMAX": "800",  # 800 mb
+            "GDAL_DISABLE_READDIR_ON_OPEN": "EMPTY_DIR",
+            "GDAL_HTTP_MERGE_CONSECUTIVE_RANGES": "YES",
+            "GDAL_HTTP_MULTIPLEX": "YES",
+            "GDAL_HTTP_VERSION": "2",
+            "PYTHONWARNINGS": "ignore",
+            "VSI_CACHE": "TRUE",
+            "VSI_CACHE_SIZE": "5000000",  # 5 MB (per file-handle)
+            "RIO_TILER_MAX_THREADS": "1" # turn off rio-tiler threading, better for lamda
+        }
+
+        # Lambda Function Definition
+        lambda_function = aws_lambda.Function(
+            self,
+            f"titiler-{stage}-TitilerFuntion",
+            runtime=aws_lambda.Runtime.PYTHON_3_8,
+            code=aws_lambda.Code.from_asset(
+                path=os.path.abspath("./"),
+                bundling=BundlingOptions(
+                    image=DockerImage.from_build(
+                        path = os.path.abspath("./"), file="resources/Dockerfile",
+                        platform = "linux/amd64"
+                    ),
+                    command=["bash", "-c", "cp -R /var/task/. /asset-output/."],
+                ),
+            ),
+            handler="handler.handler",
+            memory_size=3008,
+            timeout=Duration.seconds(600),
+            environment=env,
+        )
+
+        # TODO: Remove when staging and production no longer use lambda
```

Comment:
For these 2 TODOs: How can I assess whether they are done? Is there a ticket we can link to or some documentation or change log?

Reply:
Those were vestigal and not used. We are using a rewrite middleware instead of a lambda now. Removed.